### ISSUE TITLE
Running `osh_eval -c 'jobs'` crashes

### ISSUE THREAD

**Issue by jjbandit** [oilshell/oil#1284]
Apparently the jobs builtin either isn't hooked up, or isn't implemented.  I found this by debugging this test:

oil/_tmp/spec/cpp/background.test.html#L156

We're throwing a KeyError when trying to look up the builtin, and crashing.  Catching that error in python and printing a helpful error message for un-implemented builtins would be nice if there are more unimplemented builtins.

**Comment by andychu** [comment]
Yes good point, this is the difference between `bin/osh` and `bin/osh_eval`.  The latter is the statically typed subset that can be translated.

We have a hack to "virtually #ifdef" stuff out, and "jobs" is one of those things, which causes the KeyError

**Comment by andychu** [comment]
So yeah we need a "minimum usable oil-native release" and this will definitely be part of it

I'm thinking about the naming issue over on Zulip: https://oilshell.zulipchat.com/#narrow/stream/325160-oil-discuss-public/topic/New.20Names.20.2F.20Renaming.3F

i.e. it won't be called `oil-native` anymore

**Comment by andychu** [comment]
I found the code here:

https://github.com/oilshell/oil/blob/master/core/shell_native.py#L138

So I guess what we can do is

```
else:
  b[builtin_i.jobs] = NotImplementedBuiltin()   # shows you a nice error message
```
